### DISCUSSION TITLE
Unable to push any further updates

### DISCUSSION THREAD

**Discussion by moalamri** [moalamri/vscode-inline-fold#132] [Announcements]
Hi @emedeirox 
I’ve been battling with Microsoft security team for the past few months due to security lock on my Azure Dev account which I used for this extension.
I was able to push the latest two updates because my Azure access token was still active and valid, but now it is expired and there's no way to publish any further updates.
And according to MST my account will never be unlocked again as discussed in Service Request SIR17062913
What should we do? The only solution is to make an another account, which means a reborn version of Inline Fold extension.
Any suggestions?

Here are some of the last emails between me and MSST, it seems like there are no ways to get access to the account or to at least transfer ownership to a new one.

![image](https://github.com/moalamri/vscode-inline-fold/assets/964077/b98771b6-caf6-4184-a80d-c296bedcd39f)
--
![image](https://github.com/moalamri/vscode-inline-fold/assets/964077/8e269fdf-7563-4fe6-9a05-4479d5085204)
--
![image](https://github.com/moalamri/vscode-inline-fold/assets/964077/b71229b6-5f81-4774-9b90-4aff732fefd9)
--
![image](https://github.com/moalamri/vscode-inline-fold/assets/964077/bda1590e-e827-4c34-8a26-3c03260e20e5)

I'm truly sorry for all the hard work done by the contributors :( 
I will open a new discussion soon for a new startup.

**Comment by emedeirox** [comment]
I'm out of words...
But if there is no other way, so be it.

**Comment by moalamri** [comment]
Hi @AphLute
I see that you have forked and modified a version of the extension and used a different approach (language syntax over regex) which is a huge step forward, and perhaps should been the main method of the extension, and I think it is the best approach for a reborn version.
I want you to have a look at this topic and give your opinion on what should be our next plan if you are willing to collaborate.
Thank you so much.

**Comment by AphLute** [comment]
Hi @moalamri 
I looked at the tickets before forking your repo but didn't see this, and I'm so upset they've done this to you!

For a few years I've wanted an extension like this, but mine is based on yours completely and I wouldn't have done it without yours.  I also thought that if people used mine, we could work on a PR to merge "type folding" into yours.

I am happy to collaborate and help in any way I can with this.  If I can help with the problematic security practices while promoting your extension, I want to.  You can also contact me at aph@suncapped.com

**Reply by moalamri** [reply]
Hey @AphLute 
The security lock wasn't related to the extension, it is due to data compromising, which I can't figure why and how, but because I enabled some security features to my account it became impossible to retrieve it after it got locked by Microsoft.

Im sad to see this extension die in such early stage, but bad things happen and could be a good start for something better.

As I have suggested in this post, the only solution is to make a completely new version of this extension, which should be called "reborn version" since it will be based on the original code with the original contributors but with a different publisher. You can find many extensions with same "version" if you search "reborn" in the extensions marketplace.

I mentioned you here, because it would be a great start if we could build the new version with the same approach as you did to the forked version. What do you think?

**Reply by AphLute** [reply]
@moalamri I like this idea and I would like to help.  Your extension is great.  

Morally you're in the right and I don't like that you have to consider depending on me or anyone else to publish your extension.  Your code does the hard work to make things actually happen.  You found edge cases and solved them.  The internet brought smart people around the world together.  It's no good that we're being blocked.

I am thinking that you're proposing we do something under my account?  My only hesitation is reviewing code, and that you deserve most of the credit.  Do you think we have any chance with petitioning for your account?

The reason I separated "regex folding" from "typescript folding" is because I thought people would want the typescript functionality separately.  Even if we put the ts folding into yours, I'm not sure it would do as well as a pure "hide ts" extension.  Even for myself.  That's why I was waiting to see if anyone cared about mine before doing a reverse PR into yours with that.

That said, I'll help in any way you think will work.  I am up for doing a new version with my approach in a way that is useful to you.  Really, just tell me how I can help.

**Reply by moalamri** [reply]
Hey @AphLute 
Thank you so much for being very supportive. 
This is an open world where we can share code and take some credit while giving other people the chance to contribute and build something beautiful together. A lot of people have contributed to this extension in many different shapes and it saddens me to see that all goes for nothing, I will update this post with a screenshot from my last contact with MSST so that at least I won't take the blame for killing this extension.

I'm going to open a new discussion in two weeks (since I am currently working on some work related projects) there we could discuss the best ways to handle a new version of this extension. I appreciate that you have provided your contact information, which I might contact you there when we will be discussing anything related to accounts, since I still have the access to my npmjs account.

**Reply by tylerlaprade** [reply]
@moalamri @AphLute, where should we look for the alternative extension you mentioned?

**Reply by moalamri** [reply]
> @moalamri @AphLute, where should we look for the alternative extension you mentioned?

I haven't tried it but maybe this could do the job [https://marketplace.visualstudio.com/items?itemName=stivo.tailwind-fold](https://marketplace.visualstudio.com/items?itemName=stivo.tailwind-fold)

**Reply by bilalfawadkhan** [reply]
@moalamri  I am in love with this extension, I hope and will be following to see it go in a better direction which helps you out in getting this extension back on track. You patience is admirable , such efforts shouldn't go to waste.

**Reply by sturgeonphillip** [reply]
@moalamri and @AphLute has there been any publishing updates to the marketplace? I'm unable to find it, but would love to see what you've come up with in terms of combining your two implementations.

**Comment by mlntr** [comment]
What's your recommendation for users now @moalamri?

There is https://github.com/stivoat/tailwind-fold (although the author @stivoat doesn't seem very active)

**Reply by moalamri** [reply]
I really don't have any suggestion, I looked at the extension you shared and it's more specific unlike inline-fold which is more generic.

I guess a reborn version is the best option if we want to keep this extension alive.

**Reply by singularitti** [reply]
Have a look at https://github.com/Etsi0/class-collapse ?

**Comment by itzmesonic** [comment]
I think if you create a new acoount, clone this repo(not fork), and deploy it again its works and you can back to improving this extension

So, if you recover, please fix the bug that you can't correctly move between elements

**Oh, i forgot to say thank you for your great extension, its very helped me**

**-- One of your extension users, arsalan**

**Comment by isidorn** [comment]
Consider marking that extension as deprecated https://github.com/microsoft/vscode-discussions/discussions/1
Thanks - Isidor from the VS Code team.

**Reply by moalamri** [reply]
Is it possible to depreciate it without an alternative (new extension)?

**Reply by isidorn** [reply]
It is possible. Just comment on that thread.

**Reply by tylerlaprade** [reply]
@isidorn, is there anything you can do to escalate the errant security violation?

**Reply by isidorn** [reply]
Just comment on that thread https://github.com/microsoft/vscode-discussions/discussions/1
And we can look into deprecating

**Reply by tylerlaprade** [reply]
@isidorn, that isn't the desired outcome here. The author was flagged as violating security rules when he wasn't, leading to the extension being flagged when it shouldn't be. Ideally we want to undo the incorrect flag.

**Reply by moalamri** [reply]
@tylerlaprade I’m sorry I want to clear things up. The extension wasn’t flagged as security violating, it was the developer account in VS extension market. It was suspicious activity (unauthorized access) that triggered security lockdown to the account. I have tried so hard to gain back access but to no avail.

I don’t think that VSC team can solve this. Only MSST.

**Comment by codayon** [comment]
This extension was mentioned in a video of vs code's YouTube channel. And, that's how I know about it.
https://www.youtube.com/shorts/oVS-NPxGZmE

**Comment by sturgeonphillip** [comment]
@moalamri thanks for creating such a helpful extension.
Do we have any update on a rebirth or rebuild with @AphLute or was the idea abandoned?

**Reply by davecarrijo** [reply]
Waiting for it

**Reply by moalamri** [reply]
I might publish a new extension just for people to contribute to it and make it more beautiful.

**Reply by babakfp** [reply]
hi 👋. any updates on this?
there is this extension https://marketplace.visualstudio.com/items?itemName=stivo.tailwind-fold, it has some unique features, but missing some other, like regex. it's also not getting any updates.

**Reply by Omeshcoding** [reply]
> I might publish a new extension just for people to contribute to it and make it more beautiful.
@moalamri 
That would be great. Looking forward to the new extension

**Comment by ausrasplex** [comment]
Hi @moalamri,

I was wondering if you could tell me the theme for VS Code that was used in the screenshot?
![theme](https://github.com/user-attachments/assets/7f2f6b8e-34dc-4b73-987f-66d7722b88ac)

**Reply by davecarrijo** [reply]
nord
